pkg/kine/sqllog/sqllog.go

```diff
@@ -354,17 +363,24 @@ func (s *SQLLog) ttl(ctx context.Context) {
 	}()
 }
 
-func (s *SQLLog) Watch(ctx context.Context, key string, startRevision int64) (<-chan []*server.Event, error) {
+func (s *SQLLog) Watch(ctx context.Context, key string, startRevision int64) (server.WatchResult, error) {
 	ctx, span := otelTracer.Start(ctx, fmt.Sprintf("%s.Watch", otelName))
 	defer span.End()
 	span.SetAttributes(
 		attribute.String("key", key),
 		attribute.Int64("startRevision", startRevision),
 	)
 
+	res := make(chan []*server.Event, 100)
+	errc := make(chan error, 1)
```

Comment:
This channel is used only in the synchronous part of this request (i.e. before starting the background job that actually watches).

So, it actually has no real advantage over just returning an error from the function given that once `SQLLog.Watch` returns, no code will ever notify this channel.

Also, it is never closed, which is in general a problem.